tools/build_api.py

```diff
@@ -753,6 +755,8 @@ def build_library(src_paths, build_path, target, toolchain_name,
     try:
         res = Resources(notify).scan_with_toolchain(
             src_paths, toolchain, dependencies_paths, inc_dirs=inc_dirs)
+        res.filter(resource_filter)
+        res.filter(PsaManifestResourceFilter())
```

Comment:
I'm having trouble understanding this line. It looks like you're limiting the JSON files in the resource object to just PSA manifest files. Why is this filter being used? And why just when building libraries?

Reply:
build_library and build_project should filter files differently
as for the second filter, we need filter out the PSA manifests so they wont be copied to the build folder and cause duplication errors when building for the 2nd time

Reply:
AH, it's filtering the PSA manifests _out_, didn't realize that the first time I looked at it.